gk/main.c

```diff
@@ -546,44 +701,96 @@ gk_proc(void *arg)
 
 			/* 
 			 * Find the flow entry for the IP pair.
-			 * Create a new flow entry if not found.
+			 *
+			 * If the pair of source and destination addresses 
+			 * is in the flow table, proceed as the entry instructs,
+			 * and go to the next packet.
 			 */
 			ret = rte_hash_lookup_with_hash(
 				instance->ip_flow_hash_table,
 				&packet.flow, pkt->hash.rss);
-			if (ret < 0) {
-				/* Create a new flow entry. */
-				ret = rte_hash_add_key_with_hash(
-					instance->ip_flow_hash_table,
- 					(void *)&packet.flow, pkt->hash.rss);
-				if (ret < 0) {
-					RTE_LOG(ERR, HASH,
-						"The GK block failed to add new key to hash table!\n");
-					rte_pktmbuf_free(pkt);
+			if (ret >= 0)
+				fe = &instance->ip_flow_entry_table[ret];
+			else {
+				/*
+				 * Otherwise, look up the destination address
+			 	 * in the global LPM table.
+				 */
+				struct gk_policy *policy = get_gk_policy(
+					&gk_conf->lpm_tbl, &packet.flow);
+
+			 	/*
+				 * No entry for the destination,
+				 * drop the packet.
+				 */
+				if (policy == NULL) {
+					RTE_LOG(ERR, GATEKEEPER,
+					"gk: failed to get the policy entry for the destination!\n");
```

Comment:
Adding the source and destination addresses to the message would help operators to quickly diagnostic the problem in production.